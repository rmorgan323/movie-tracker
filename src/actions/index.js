import { getRecentMovies } from '../helper/getRecentMovies/getRecentMovies';
import addNewUser from '../helper/addNewUser/addNewUser';
import getUser from '../helper/getUser/getUser';
import checkUser from '../helper/checkUser/checkUser';
import getUserFavorites from '../helper/getUserFavorites/getUserFavorites';
import postNewFavoriteData from '../helper/postNewFavoriteData/postNewFavoriteData';
import deleteFavoriteData from '../helper/deleteFavoriteData/deleteFavoriteData';

export const makeMovieArray = movies => ({
  type: 'MAKE_MOVIE_ARRAY',
  movies
});

export const getMovies = () => async dispatch => {
  const newMovies = await getRecentMovies();
  dispatch(makeMovieArray(newMovies));
};

export const newUser = user => ({
  type: 'ADD_NEW_USER',
  user
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const fetchUsers = user => async dispatch => {
  const response = await addNewUser(user);
  if (response.status === 'success') {
    const user = await getUser(response.id);
    dispatch(setUser(user));
  } else {
    dispatch(signupFailure('error-signup'));
  }
};

export const signupFailure = msg => ({
  type: 'SIGNUP_ERROR',
  msg
});

export const checkForUser = user => async dispatch => {
  const response = await checkUser(user);
  if (response.status === 'success') {
    dispatch(setUser(response.data));
  } else {
    dispatch(signupFailure('error-login'));
  }
};

export const userLogout = () => ({
  type: 'USER_LOGOUT'
});

export const deletePost = (movie) => async dispatch => {
  const response = await deleteFavoriteData(movie.userId, movie.movieId);

  if (response.status === 'success') {
    dispatch(deleteFavorite(movie));
  }
};

export const deleteFavorite = (movie) => ({
  type: 'DELETE_FAVORITE',
  movie
});

export const addPost = (movie) => async dispatch => {
  const response = await postNewFavoriteData(movie);

  if (response.status === 'success') {
    dispatch(addFavorite(movie));
  }
};

export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});

export const checkUserFavorites = (userId) => async dispatch => {
  const response = await getUserFavorites(userId);
  dispatch(setUserFavorites(response.data));
};

export const setUserFavorites = (favorites) => ({
  type: 'SET_USER_FAVORITES',
  favorites
});

