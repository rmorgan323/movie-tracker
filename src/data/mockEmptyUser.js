export const mockEmptyUser = {
  signedIn: false,
  userInfo: {
    id: '',
    name: '',
    password: '',
    email: '',
    favorites: []
  }
};
