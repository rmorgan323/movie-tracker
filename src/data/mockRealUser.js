import { mockMovie } from './mockMovie';

export const mockRealUser = {
  signedIn: true,
  userInfo: {
    id: 1,
    name: 'Batman',
    password: 'darkKn1ght',
    email: 'supermanSux@batmanrulez.com',
    favorites: [mockMovie]
  }
};
