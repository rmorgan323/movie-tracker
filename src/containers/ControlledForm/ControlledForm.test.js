import {
  ControlledForm,
  mapStateToProps,
  mapDispatchToProps
} from './ControlledForm';
import React from 'react';
import { shallow } from 'enzyme';
import mockEmptyUser from '../../data/mockEmptyUser';
import mockRealUser from '../../data/mockRealUser';
import mockLocationData from '../../data/mockLocation';
import * as actions from '../../actions';

const defaultUserState = {
  name: '',
  password: '',
  email: ''
};

const defaultErrorState = '';

describe('ControlledForm tests', () => {
  let renderedControlledForm;
  let mockLocation = mockLocationData;

  beforeEach(() => {
    renderedControlledForm = shallow(
      <ControlledForm user={mockEmptyUser} location={mockLocation} />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedControlledForm).toMatchSnapshot();
  });

  it('Should have a default user state and a default error message state', () => {
    expect(renderedControlledForm.state('users')).toEqual(defaultUserState);
    expect(renderedControlledForm.state('errorMessage')).toEqual(
      defaultErrorState
    );
  });

  it('Should render an error message if one is in state', () => {
    const signupErrorMessage =
      'That email is already registered. Please use another email or try logging in.';

    expect(renderedControlledForm.find('.error-message').text()).toEqual('');

    renderedControlledForm.setState({ errorMessage: signupErrorMessage });

    expect(renderedControlledForm.find('.error-message').text()).toEqual(
      signupErrorMessage
    );
  });

  it('Should have an extra input if endpoint is /signup', () => {
    expect(renderedControlledForm.find('input').length).toEqual(2);

    mockLocation.pathname = '/signup';
    renderedControlledForm = shallow(
      <ControlledForm user={mockEmptyUser} location={mockLocation} />
    );

    expect(renderedControlledForm.find('input').length).toEqual(3);
  });

  describe('mapStateToProps tests', () => {
    it('Should pull a user from store', () => {
      const mockStore = {
        user: mockRealUser
      };
      const result = mapStateToProps(mockStore);

      expect(result.user).toEqual(mockStore.user);
    });
  });

  describe('mapDispatchToProps tests', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    xit('Should call dispatch when getUsers is called', () => {
      result.getUsers();

      expect(mockDispatch).toHaveBeenCalledWith(actions.fetchUsers());
    });

    xit('Should call dispatch when checkUsers is called', () => {
      result.checkUsers();

      expect(mockDispatch).toHaveBeenCalledWith(actions.checkForUser());
    });

    xit('Should call dispatch when getUserFavorites is called', () => {
      result.getUserFavorites();

      expect(mockDispatch).toHaveBeenCalledWith(actions.checkUserFavorites());
    });
  });
});
