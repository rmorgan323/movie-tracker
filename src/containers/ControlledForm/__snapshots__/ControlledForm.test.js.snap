// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledForm tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlledForm
    location={
        Object {
            "hash": "",
            "pathname": "",
            "search": "",
            "state": undefined,
          }
    }
    user={
        Object {
            "signedIn": false,
            "userInfo": Object {
              "email": "",
              "favorites": Array [],
              "id": "",
              "name": "",
              "password": "",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-button-container"
>
        <div
                className="login-buttons"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login-button form-button"
                        to="/login"
                >
                        Login
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login-button form-button"
                        to="/signup"
                >
                        Sign-up
                </NavLink>
        </div>
        <form
                onSubmit={[Function]}
        >
                <input
                        className="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
                />
                <input
                        className="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                />
                <button
                        className="submit-button"
                        onClick={[Function]}
                        type="submit"
                >
                        Submit
                </button>
        </form>
        <div
                className="error-message"
        />
</div>,
      "className": "form-component",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="login-buttons"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login-button form-button"
                        to="/login"
            >
                        Login
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login-button form-button"
                        to="/signup"
            >
                        Sign-up
            </NavLink>
</div>,
          <form
            onSubmit={[Function]}
>
            <input
                        className="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
            />
            <input
                        className="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        className="submit-button"
                        onClick={[Function]}
                        type="submit"
            >
                        Submit
            </button>
</form>,
          <div
            className="error-message"
/>,
        ],
        "className": "form-button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="login-button form-button"
                to="/login"
>
                Login
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="login-button form-button"
                to="/signup"
>
                Sign-up
</NavLink>,
            ],
            "className": "login-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Login",
                "className": "login-button form-button",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Sign-up",
                "className": "login-button form-button",
                "to": "/signup",
              },
              "ref": null,
              "rendered": "Sign-up",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <input
                className="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
/>,
              <input
                className="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                className="submit-button"
                onClick={[Function]}
                type="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "submit-button",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "error-message",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-button-container"
>
          <div
                    className="login-buttons"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="login-button form-button"
                              to="/login"
                    >
                              Login
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="login-button form-button"
                              to="/signup"
                    >
                              Sign-up
                    </NavLink>
          </div>
          <form
                    onSubmit={[Function]}
          >
                    <input
                              className="email"
                              onChange={[Function]}
                              placeholder="Email"
                              type="text"
                              value=""
                    />
                    <input
                              className="password"
                              onChange={[Function]}
                              placeholder="Password"
                              type="password"
                              value=""
                    />
                    <button
                              className="submit-button"
                              onClick={[Function]}
                              type="submit"
                    >
                              Submit
                    </button>
          </form>
          <div
                    className="error-message"
          />
</div>,
        "className": "form-component",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="login-buttons"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="login-button form-button"
                            to="/login"
              >
                            Login
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="login-button form-button"
                            to="/signup"
              >
                            Sign-up
              </NavLink>
</div>,
            <form
              onSubmit={[Function]}
>
              <input
                            className="email"
                            onChange={[Function]}
                            placeholder="Email"
                            type="text"
                            value=""
              />
              <input
                            className="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
              />
              <button
                            className="submit-button"
                            onClick={[Function]}
                            type="submit"
              >
                            Submit
              </button>
</form>,
            <div
              className="error-message"
/>,
          ],
          "className": "form-button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="login-button form-button"
                  to="/login"
>
                  Login
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="login-button form-button"
                  to="/signup"
>
                  Sign-up
</NavLink>,
              ],
              "className": "login-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Login",
                  "className": "login-button form-button",
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Sign-up",
                  "className": "login-button form-button",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "Sign-up",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                <input
                  className="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
/>,
                <input
                  className="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
/>,
                <button
                  className="submit-button"
                  onClick={[Function]}
                  type="submit"
>
                  Submit
</button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "submit-button",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "error-message",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
